core/appmodule/v2/handlers.go

```diff
@@ -44,7 +44,11 @@ type PostMsgRouter interface {
 	// the provided name.
 	Register(msgName string, handler PostMsgHandler)
 	// RegisterGlobal will register a global message handler hooking after the execution of any message.
-	RegisterGlobal(handler PreMsgHandler)
+	RegisterGlobal(handler PostMsgHandler)
```

Comment:
Should we not update those APIs instead so we don't need a wrapper around the router builder?
Additionally it would match better RFC 006.

Reply:
updated